From 29c186c05783997748962ae13284a80244ac3d84 Mon Sep 17 00:00:00 2001
From: c4pp4
Date: Wed, 21 Oct 2026 16:18:15 +0200
Subject: [PATCH 1/1] Use ubuntu help when in Unity

Open the screenshot page of the Ubuntu help instead of the MATE user
guide when running in Unity.

Signed-off-by: c4pp4
---
 mate-screenshot/src/mate-screenshot.c | 34 +++++++++++++++++++++++----
 1 file changed, 30 insertions(+), 4 deletions(-)

diff --git a/mate-screenshot/src/mate-screenshot.c b/mate-screenshot/src/mate-screenshot.c
index 9786c99..6eef08a 100644
--- a/mate-screenshot/src/mate-screenshot.c
+++ b/mate-screenshot/src/mate-screenshot.c
@@ -122,15 +122,41 @@ static GtkWidget *delay_hbox = NULL;
 
 void loop_dialog_screenshot (void);
 
+static gboolean
+in_desktop (const gchar *name)
+{
+  const gchar *desktop_name_list;
+  gchar **names;
+  gboolean in_list = FALSE;
+  gint i;
+
+  desktop_name_list = g_getenv ("XDG_CURRENT_DESKTOP");
+  if (!desktop_name_list)
+    return FALSE;
+
+  names = g_strsplit (desktop_name_list, ":", -1);
+  for (i = 0; names[i] && !in_list; i++)
+    in_list = g_strcmp0 (names[i], name) == 0;
+  g_strfreev (names);
+
+  return in_list;
+}
+
 static void
 display_help (GtkWindow *parent)
 {
   GError *error = NULL;
 
-  gtk_show_uri_on_window (parent,
-                          "help:mate-user-guide/goseditmainmenu-53",
-                          gtk_get_current_event_time (),
-                          &error);
+  if (in_desktop ("Unity"))
+    gtk_show_uri_on_window (parent,
+                            "help:ubuntu-help/screen-shot-record",
+                            gtk_get_current_event_time (),
+                            &error);
+  else
+    gtk_show_uri_on_window (parent,
+                            "help:mate-user-guide/goseditmainmenu-53",
+                            gtk_get_current_event_time (),
+                            &error);
 
   if (error)
     {
-- 
2.55.0

